IdentityCore/src/cache/ios/MSIDKeychainTokenCache.m

```diff
@@ -625,69 +707,10 @@ - (BOOL)clearWithContext:(id<MSIDRequestContext>)context
     return YES;
 }
 
-- (BOOL)saveAppMetadata:(MSIDAppMetadataCacheItem *)item
-                    key:(MSIDCacheKey *)key
-             serializer:(id<MSIDAppMetadataItemSerializer>)serializer
-                context:(id<MSIDRequestContext>)context
-                  error:(NSError **)error
-{
-    assert(item);
-    assert(serializer);
-    
-    NSData *itemData = [serializer serializeAppMetadataCacheItem:item];
-    
-    if (!itemData)
-    {
-        if (error)
-        {
-            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Failed to serialize app metadata item.", nil, nil, nil, context.correlationId, nil);
-        }
-        MSID_LOG_ERROR(context, @"Failed to serialize app metadata item.");
-        return NO;
-    }
-    
-    MSID_LOG_INFO_PII(context, @"Save keychain item, item info %@", item);
-    
-    return [self saveData:itemData
-                      key:key
-                  context:context
-                    error:error];
-}
-
-- (NSArray<MSIDAppMetadataCacheItem *> *)appMetadataEntriesWithKey:(MSIDCacheKey *)key
-                                                        serializer:(id<MSIDAppMetadataItemSerializer>)serializer
-                                                           context:(id<MSIDRequestContext>)context
-                                                             error:(NSError **)error
+// Override the following function in subclasses if special key handling is needed
+- (MSIDCacheKey *)tokenCacheKeyFromCacheKey:(MSIDCacheKey *)key
```

Comment:
I am a bit concerned that we do this in low level (data source). In my opinion, datasource is just a container, like Dictionary, DB, plist etc. Some class on top of it should know how to create storing keys (MSIDDefaultTokenCacheAccessor, MSIDLegacyTokenCacheAccessor).

Reply:
I'd consider broker key being more like low level separation of cache per app, like keychain access groups. So, from model perspective, it belongs to the low level cache more than to accessor level. 

Having said that, I'm not sure we need to operate on cache key level in here, vs on keychain query level. Let me check the code, and provide suggestions if necessary.

Reply:
"keychain access groups" is something unique for this specific datasource, it is just a configuration needed for this specific datasource to work, like password & username for SQL database. When overriding key behaviour will affect all datasources. Right now we have only 2 datasource (ios keychain, mac in-memory), but if we assume that we have 5 datasources. With current logic we would need to modify all 5 datasources to make it work, instead of applying this logic on cache level. Anyways, it is just my opinion, don't insist on changes here.

Reply:
Not sure I follow the logic of 5 data sources. Let's sync offline.

Reply:
I don't think it should it an accessor level, this seems accessor independent.

Reply:
Here is my opinion :)
Broker needs an isolation mechanism. I.e., AT/RT/FRT stored by app1 won't be found by app2 if they are using different app symmetric keys. Such that we have an idea of making the app symmetric key being part of the keychain key, so serve such a purpose.
The following two points make me think making changes at keychain cache level is more appropreate:
- The app symmetric key is specific to a keychain cache instance;
- Only keychain data source needs isolation, but not other data sources, like mac cache.

But I am open to any opinion :)